backends/bmv2/run-bmv2-test.py

```diff
@@ -51,9 +51,12 @@ def __init__(self):
         self.verbose = False
         self.replace = False            # replace previous outputs
         self.compilerOptions = []
+        self.switchOptions = []
```

Comment:
Overall, I think it would be a lot cleaner if you could drive your test case with a single shell script, as opposed to adding three new arguments to run-bmv2-test.py to forward command options to bmv2 switch and to invoke make.

Reply:
At a first glance, the shell script would only allow me to remove the --make option. I still need to pass arguments to simple_switch. Either I still add the arguments via `run-bmv2-test.py`, or I disable that part of the code and do it manually, or bypass `run-bmv2-test.py` altogether.

I think this way is more reusable, and there's less code duplication.

Reply:
Instead of extending run-bmv2-test.py to support invoking make, how about we extend the python script with an init script/step/action before executing the bmv2 switch. In your test case, the init script will call `make` and built a shared library. In other future test cases, the init script can call other linux utility tools. The run-bmv2-test.py script does not need to know the details of each init script and it certainly need not to know how to invoke `make`.